treeopt/treeOpt.py

```diff
@@ -0,0 +1,277 @@
+        :param filename: Name of the file
+        :type filename: String
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        Path(os.getcwd() + "/treeOptData").mkdir(parents=True, exist_ok=True)
+        path = os.path.join(os.getcwd() + "/treeOptData", filename + ".csv")
+        np.savetxt(path, npArray, delimiter=",")
+
+    def read_data(self, filename):
+        """
+        Function that reads a file in the treeOptData directory and creates a
+        numpy array with this data
+        :param filename: Name of the file to be read
+        :type filename: String
+        :return: Array containing the information of the file
+        :rtype: Numpy Array
+
+        """
+
+        path = os.path.join(os.getcwd() + "/treeOptData", filename + ".csv")
+        array = np.loadtxt(path, delimiter=",")
+        return array
+
+    # Set functions (to set functions/methods/variables prior to optimizaion)
+    def set_name(self, name):
+        """
+        Sets a Variable with the name of the Optimization Problem
+        :param name: the Name of the Optimization problem
+        :type name: String
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.name = name
+
+    def set_sampling_method(self, method):
+        """
+        Sets the Sampling Method inteddet to be used by the individual
+        Optimization Problem
+        :param method: Python Object
+        :type method: TYPE
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.samplingMethod = method
+
+    def set_num_doe(self, numDOE):
+        """
+        Sets the Number of Sampling Points to be used in the initial Sampling
+        of the design space
+        :param numDOE: Number of Sampling Points
+        :type numDOE: Integer
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.numDOE = numDOE
+
+    def set_limits(self, limits):
+        """
+        Sets the search limits of each dimension of the design space
+        :param limits: Array containing the highest and lowest limits for each
+        dimension
+        :type limits: Numpy-Array
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.limits = limits
+
+    def set_simulate_method(self, method):
+        """
+        Sets the method, which should be used to start the Simulation
+        :param method: A keyword representing the method
+        the simulation String
+        :type method: Python function
+        :return: Nothing
+        :rtype: None
+
+        """
+        self.simKeyword = method
+        if self.simKeyword == "benchmarking":
+            self.simulateMethod = simulate.simulate_benchmark_function
+        elif self.simKeyword == "python":
+            self.simulateMethod = simulate.simulate_benchmark_function
+        elif self.simKeyword == "extern":
+            self.simulateMethod = simulate.simulateExternalProgramm
+        else:
+            print("Error")
```

Comment:
Two things:
 - Please be more specific in the error message.
 - Don't print error message without proper error handling. Please throw a `NotImplementedError` here.